model_zoo/census_model_sqlflow/wide_and_deep/wide_deep_functional_keras.py

```diff
@@ -0,0 +1,299 @@
+import tensorflow as tf
+
+from elasticdl.python.elasticdl.callbacks import LearningRateScheduler
+from elasticdl.python.keras.layers.sparse_embedding import SparseEmbedding
+from elasticdl_preprocessing.layers.concatenate_with_offset import (
+    ConcatenateWithOffset,
+)
+from elasticdl_preprocessing.layers.discretization import Discretization
+from elasticdl_preprocessing.layers.hashing import Hashing
+from elasticdl_preprocessing.layers.index_lookup import IndexLookup
+from elasticdl_preprocessing.layers.to_sparse import ToSparse
+from model_zoo.census_model_sqlflow.wide_and_deep.feature_configs import (
+    FEATURE_TRANSFORM_INFO_EXECUTE_ARRAY,
+    INPUT_SCHEMAS,
+    TRANSFORM_OUTPUTS,
+    age_bucketize,
+    capital_gain_bucketize,
+    capital_loss_bucketize,
+    education_hash,
+    group1,
+    group1_embedding_deep,
+    group1_embedding_wide,
+    group2,
+    group2_embedding_deep,
+    group2_embedding_wide,
+    group3,
+    group3_embedding_deep,
+    hours_per_week_bucketize,
+    marital_status_lookup,
+    native_country_hash,
+    occupation_hash,
+    race_lookup,
+    relationship_lookup,
+    sex_lookup,
+    workclass_lookup,
+)
+from model_zoo.census_model_sqlflow.wide_and_deep.transform_ops import (
+    TransformOpType,
+)
+
+
+# The model definition from model zoo. It's functional style.
+# Input Params:
+#   input_layers: The input layers dict of feature inputs
+#   wide_embeddings: The embedding list for the wide part
+#   deep_embeddings: The embedding list for the deep part
+def wide_and_deep_classifier(input_layers, wide_embeddings, deep_embeddings):
+    # Wide Part
+    wide = tf.keras.layers.Concatenate()(wide_embeddings)  # shape = (None, 3)
+
+    # Deep Part
+    dnn_input = tf.reshape(deep_embeddings, shape=(-1, 3 * 8))
+    for i in [16, 8, 4]:
+        dnn_input = tf.keras.layers.Dense(i)(dnn_input)
+
+    # Output Part
+    concat_input = tf.concat([wide, dnn_input], 1)
+
+    logits = tf.reduce_sum(concat_input, 1, keepdims=True)
+    probs = tf.reshape(tf.sigmoid(logits), shape=(-1,))
+
+    return tf.keras.Model(
+        inputs=input_layers,
+        outputs={"logits": logits, "probs": probs},
+        name="wide_deep",
+    )
+
+
+# Build the input layers from the schema of the input features
+def get_input_layers(input_schemas):
+    input_layers = {}
+
+    for schema_info in input_schemas:
+        input_layers[schema_info.name] = tf.keras.layers.Input(
+            name=schema_info.name, shape=(1,), dtype=schema_info.dtype
+        )
+
+    return input_layers
+
+
+# Build the transform logic from the metadata in feature_configs.py.
+def transform(inputs):
+    transformed = inputs.copy()
```

Comment:
Why do you copy `inputs` to transformed not `transformed = inputs`?

Reply:
If we don't add copy() here. 
We will update the input layers directly.
1. The value of the existed key will be updated: such as `age: <tensorflow.python.framework.sparse_tensor.SparseTensor object at 0x13b237a90>`
2. More keys will be added in the input layers such as age_bucketize.

And then the updated input layers will be fed into the keras.Model(inputs=inputs, xxx), the input layers for the model is not the origin input features.